### PR TITLE
Update the charts if the number of queries has increased significantly

### PR DESCRIPTION
**Pull request by DL6ER**
# What does this implement/fix?

This PR addresses the issue mentioned in https://github.com/pi-hole/FTL/pull/1977#issuecomment-2191035021

---

**Related issue or feature (if applicable):** https://github.com/pi-hole/FTL/pull/1977

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

### CODE REVIEW

**Comment by yubiuser** [comment]
Good idea - but it's not working for me right now. After the queries are imported the count of 'Total queries' jumps up from 0 to a few thousand but no charts are updated. After a while the top lists are updated/populated but the 'overTime' graphs are not (or I did not wait long enough)

**Comment by DL6ER** [comment]
The update should work immediately so not having waited long enough isn't an option. I looked again at the logic and you description of

> the count of 'Total queries' jumps up from 0 to a few thousand

translates into

``` js
if (0 < 1000 && 1000 > 100) {
  // update the charts
}
```

should definitely have fired. Either this is a caching issue with the JS file or - I'm afraid - I have to ask you to do some debugging on this when time allows.

**Comment by yubiuser** [comment]
Ok, you have probably be right with the caching issue. I tried with a fresh cache on FF and Chrome and it worked better - but now I get a constant re-drawing. I think we miss an exit of the loop somewhere.

Peek 2024-06-28 21-16

File: scripts/pi-hole/js/index.js [hunk 1]
```diff
@@ -394,10 +394,16 @@ function updateTopLists() {
   utils.setTimer(updateTopLists, REFRESH_INTERVAL.top_lists);
 }
 
+var previousCount = 0;
 function updateSummaryData(runOnce = false) {
   $.getJSON("/api/stats/summary", function (data) {
     var intl = new Intl.NumberFormat();
-    $("span#dns_queries").text(intl.format(parseInt(data.queries.total, 10)));
+    const newCount = parseInt(data.queries.total, 10);
+    if (previousCount === 0) {
```
Comment (yubiuser) [inline scripts/pi-hole/js/index.js]: The issue is here. During startup, if not all queries are loaded completely, `previousCount` might be >0 but `< 1/2 newCount`. But `previousCount` is never updated again, always triggering the update of the chart

Screenshot at 2024-06-28 21-49-52

___

One solution might be just to remove the conditional `if (previousCount === 0)` and always set `previousCount = newCount;`

Comment (rdwebdesign) [reply scripts/pi-hole/js/index.js]: Or add `previousCount = newCount;` in the other `if` (lines 417 to 424).

Comment (yubiuser) [reply scripts/pi-hole/js/index.js]: You're correct. If adding it before the comparison check it will never trigger the update....

Comment (DL6ER) [reply scripts/pi-hole/js/index.js]: Yes, it should be set *after* the `if`.

**Review by yubiuser — APPROVED** [review]

**Pull request merged**